File: nave-backend/src/main/java/reto/sofka2022/navebackend/modelos/Lanzadera.java
package reto.sofka2022.navebackend.modelos;

import lombok.Data;
import lombok.Setter;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AccessLevel;

@Document(collection = "tb-Lanzadera")
@Data
public class Lanzadera {

    @Id
    @Setter(AccessLevel.NONE) //le establezco acces none para que no pueda modificarse el id
    
    private String _id;
    private int potencia;
    private int toneladasEmpuje;
    private int toneladasPeso;
    private int altura;
    private int capacidadToneladas;
    @DBRef private Nave nave;
    
}

File: nave-frontEnd/src/app/modelo/lanzadera.ts

export class Lanzadera {

    _id: string;
    potencia: string;
    toneladasEmpuje: string;
    toneladasPeso: string;
    altura: string;
    capacidadToneladas: string;
    nave: string;
   



    constructor(potencia: string, toneladasEmpuje: string, toneladasPeso: string, altura: string, 
                capacidadToneladas: string, _id: string, nave: string ){
        this.potencia = potencia;
        this.toneladasEmpuje = toneladasEmpuje;
        this.toneladasPeso = toneladasPeso;
        this.altura = altura;
        this.capacidadToneladas = capacidadToneladas;
        this._id = _id;
        this.nave = nave;
        console.log("nave")
        
    }

}
